src/GenHDF5IO.chpl

```diff
@@ -0,0 +1,2193 @@
+module GenHDF5IO {
```

Comment:
Maybe we want to add the `Msg` prefix to this module (and the Parquet module) to be more consistent with the other module names now that we have added the message processing functions to these modules?